src/Shared/Microsoft.DotNet.Scaffolding.Shared/T4Templating/TemplateInvoker.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Scaffolding.Shared.T4Templating
+{
+    /// <summary>
+    /// Contains useful helper functions for running visual studio text transformation.
+    /// For internal microsoft use only. Use <see cref="ITemplateInvoker"/>
+    /// in custom code generators.
+    /// </summary>
+    public class TemplateInvoker : ITemplateInvoker
+    {
+        private readonly ConsoleLogger _consoleLogger;
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public TemplateInvoker(ConsoleLogger consoleLogger = null)
+        {
+            _consoleLogger = consoleLogger ?? new ConsoleLogger();
+        }
+
+        /// <summary>
+        /// Executes a code generator template to generate the code.
+        /// </summary>
+        /// <param name="template">ITextTransformation template object</param>
+        /// <param name="templateParameters">Parameters for the template.
+        /// These parameters can be accessed in text template using a parameter directive.
+        /// The values passed in must be either serializable or 
+        /// extend <see cref="MarshalByRefObject"/> type.</param>
+        /// <returns>Generated code if there were no processing errors. Throws 
+        /// <see cref="InvalidOperationException" /> otherwise.
+        /// </returns>
+        public string InvokeTemplate(ITextTransformation template, IDictionary<string, object> templateParameters)
+        {
+            templateParameters ??= new Dictionary<string, object>();
```

Comment:
>templateParameters [](http://example.com/codeflow?start=12&length=18)

why new dictionary? Just don't go to the loop if it is null

Reply:
fixed ty, was trying to avoid a nested if I guess.